R/slice.R

```diff
@@ -165,6 +174,81 @@ vec_as_index <- function(i, n, names = NULL) {
   vec_assert(n, integer(), 1L)
   .Call(vctrs_as_index, i, n, names)
 }
+#' @rdname vec_as_index
+#' @export
+vec_as_position <- function(i, n, names = NULL) {
+  if (is.object(i)) {
+    if (vec_is_coercible(i, int())) {
+      i <- vec_cast(i, int())
+    } else if (vec_is_coercible(i, chr())) {
+      i <- vec_cast(i, chr())
+    }
+  } else if (typeof(i) == "double") {
+    i <- vec_coercible_cast(i, int())
+  }
+
+  type <- typeof(i)
+  if (!type %in% c("integer", "character")) {
+    stop_position_incompatible_type(i)
+  }
+  if (length(i) != 1L) {
+    stop_position_incompatible_size(i)
+  }
+  if (type == "integer" && i < 1L) {
```

Comment:
Do we need to check for `NA_integer_` here too? Or does `vec_as_index()` already handle that?

Reply:
oops, I will make these OOB errors as we discussed on Slack.